Automattic/pocket-casts-android
# Profile bookmarks - Display & Sort

## Description
This displays bookmarks on the Profile screen and allows sorting them.

## Testing Instructions

1. Start the app with an user that has Plus or Patron active
3. Go to profile
4. Scroll down and check if you see a Bookmarks option
5. Tap on Bookmarks
6. Check if the list of your bookmarks is seen ( if you never set a bookmark you should see an empty view that says how to set it)
7. Tap the bookmark play button and check if it starts playing the bookmark episode at the bookmark timestamp
8. Check if sort option works
    - Tap on the more button on the top right
    - Tap on Sort By
    - Switch between different sort orders and check if it works correctly
9. Check if delete works
     - Tap on more button again
     - Tap on select bookmarks
     - Select some bookmarks
     - Tap on the delete icon ( trash button)
     - Confirm the deletion
     - Check that the delete happens
10. Check if edit works
     - Tap on more button again
     - Tap on select bookmarks
     - Select one bookmark
     - Tap on the edit button
     - Change the name of the bookmark
     - Confirm
     - Check if the name updates correctly
11. Check if share works
     - Tap on more button again
     - Tap on select bookmarks
     - Select one bookmark
     - Tap on the share button
     - Share bookmark as per steps in https://github.com/Automattic/pocket-casts-android/pull/2022#top
12. Check event tracking
    - Interact with bookmarks on the profile (change name, delete)
    - Check if tracks events have the `source` parameters set to `profile`  

### Test without Plus

1. Start the app with an user without Plus or Patron
2. Go to profile
3. Scroll down and check if you see a Bookmarks option
4. Tap on Bookmarks
5. Check if it shows a text that allows to redirect to the Plus purchase flow

## Screenshots or Screencast 
![out](https://github.com/Automattic/pocket-casts-android/assets/1405144/0e233d42-4a4f-4560-bd06-28ac9144da48)


35. ## Checklist
 
#### I have tested any UI changes...
